From 1980a8f4779a955e73285e7a0d86549b69bea5c8 Mon Sep 17 00:00:00 2001
From: Yu Zongchun <b40527@freescale.com>
Date: Sun, 28 Apr 2013 14:39:22 +0800
Subject: [PATCH] Add the compile and install rules for cryptodev tests folder

This is required to install the cryptodev tests folder to rootfs

Signed-off-by: Yu Zongchun <b40527@freescale.com>

Upstream-Status: Pending

---
 Makefile       |    6 ++++++
 tests/Makefile |    8 ++++++++
 2 files changed, 14 insertions(+), 0 deletions(-)

Index: git/Makefile
===================================================================
--- git.orig/Makefile
+++ git/Makefile
@@ -35,6 +35,9 @@ modules_install:
 	$(MAKE) $(KERNEL_MAKE_OPTS) modules_install
 	install -m 644 -D crypto/cryptodev.h $(DESTDIR)/$(includedir)/crypto/cryptodev.h
 
+install_tests:
+	make -C tests install DESTDIR=$(PREFIX)
+
 clean:
 	$(MAKE) $(KERNEL_MAKE_OPTS) clean
 	rm -f $(hostprogs) *~
@@ -43,6 +46,9 @@ clean:
 check:
 	CFLAGS=$(CRYPTODEV_CFLAGS) KERNEL_DIR=$(KERNEL_DIR) $(MAKE) -C tests check
 
+testprogs:
+	KERNEL_DIR=$(KERNEL_DIR) make -C tests testprogs
+
 CPOPTS =
 ifneq ($(SHOW_TYPES),)
 CPOPTS += --show-types
Index: git/tests/Makefile
===================================================================
--- git.orig/tests/Makefile
+++ git/tests/Makefile
@@ -23,6 +23,12 @@ bindir = $(execprefix)/bin
 
 all: $(hostprogs)
 
+install:
+	install -d  $(DESTDIR)/usr/bin/tests_cryptodev
+	for bin in $(hostprogs); do \
+		install -m 755 $${bin} $(DESTDIR)/usr/bin/tests_cryptodev/; \
+	done
+
 check: $(hostprogs)
 	./cipher
 	./hmac
@@ -38,6 +44,8 @@ install:
 		install -m 755 $$prog $(DESTDIR)/$(bindir); \
 	done
 
+testprogs: $(hostprogs)
+
 clean:
 	rm -f *.o *~ $(hostprogs)
 
